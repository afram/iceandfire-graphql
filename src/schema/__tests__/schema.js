import { graphql } from 'graphql';
import gotSchema from '..';

describe('Schema', () => {
  test('Gets an error when ID is omitted', async () => {
    const query = `{ book { name } }`;
    const result = await graphql(gotSchema, query);
    expect(result.errors.length).toBe(1);
    expect(result.errors[0].message).toBe('must provide id or bookID');
    expect(result.data).toEqual({ book: null });
  });

  test('Gets an error when global ID is invalid', async () => {
    const query = `{ book(id: "notanid") { name } }`;
    const result = await graphql(gotSchema, query);
    expect(result.errors.length).toBe(1);
    expect(result.errors[0].message).toMatchSnapshot();
    expect(result.data).toEqual({ book: null });
  });
});
