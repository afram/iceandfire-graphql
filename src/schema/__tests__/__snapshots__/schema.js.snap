// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema Gets an error when global ID is invalid 1`] = `"No entry in local cache for http://anapioficeandfire.com/api/books/��Z�'/"`;
